home-assignments/session2/cli.py

```
@@ -0,0 +1,108 @@
+        days_digit = split_str[-1]
+        if days_digit.isdigit():
+            days = int(days_digit) + 1
+
+    if days == 0:
+        print('Invalid value in period: expected TODAY+n. Defaulted to TODAY')
+        days = 1
+
+    return days
+
+
+@click.command()
+@click.argument('city', nargs=-1)
+@click.option('--period', default='TODAY', help='period in format TODAY+n')
+@click.option('-f', is_flag=True, help='-f for Fahrenheit')
+def parse_command(city, period, f):
+    city = ' '.join(city)
+    return city, period, f
+
+
+def main():
+    city, period, unparsed_unit = parse_command(standalone_mode=False)
+    unit, unit_full_name = parse_unit(unparsed_unit)
+    days_in_forecast = parse_period(period)
+
+    try:
+        weather = Weather(unit)
+    except AttributeError:
+        print('Could not load weather')
+        return
+
+    try:
+        forecasts = weather.lookup_by_location(city)
+    except requests.exceptions.HTTPError:
+        print('Error getting weather forecast')
+        return
+
+    day_today = True
+    days_counter = 0
+    try:
+        for day in forecasts.forecast[:days_in_forecast]:
+            if day_today:
+                date_to_print = 'Today'
+            else:
+                date_to_print = 'On ' + day.date
+
+            print(
+            '{date} the weather in {city} is {condition} with temperature trailing from {low_temp} to {high_temp} in {unit}'.
+                format(date=date_to_print,
+                       city=city,
+                       condition=day.text.lower(),
+                       low_temp=day.low,
+                       high_temp=day.high,
+                       unit=unit_full_name))
+            days_counter += 1
+
+            if day_today and (days_in_forecast-1) != 0:
+                print()
+                print('Forecast for the next {} days)'.format(days_in_forecast-1))
+                print()
+                day_today = False
+    except KeyError:
```

You did this a lot of time using many Try/except blocks so I would summarize here:
It is bad practice to [fail silently](https://realpython.com/the-most-diabolical-python-antipattern/) and if possible you should use as little try/except blocks as possible if there is not a real justification.

---

Agree with you. It actually was the requirment made by Arieh last assignment to use try/except blocks everywhere possible.